### ISSUE TITLE
Broken in Safari

### ISSUE THREAD

**Issue by bdougherty** [ArthurHeitmann/photon-reddit#1]
Unfortunately I can't give you a line number because it doesn't have one in the console, but here's the error:

```
[Error] SyntaxError: Invalid regular expression: invalid group specifier name
	promiseReactionJob
```

Using the latest Safari 14.1.2.

**Comment by bdougherty** [comment]
Looks to be the lookbehind stuff here: https://github.com/ArthurHeitmann/photon-reddit/blob/b6b59aab98eb2796037d86929c9770f161678479/src/static/scripts/api/redditApi.ts#L74-L77

**Comment by ArthurHeitmann** [comment]
Unfortunately I can only support Chrome and Firefox (only have Windows). To support Safari many things would need to be changed.

**Comment by ElijahPepe** [comment]
https://caniuse.com/js-regexp-lookbehind Safari, as well as some mobile browsers don't support lookbehind, which causes an issue for some users.

Instead of lookbehinds, could non-captured groups can be used instead to ensure compatibility? `/(?<=^\/)` wouldn't work, but `/(?:\/)` could because it's only extracting that first group.

**Comment by ArthurHeitmann** [comment]
Ok so I now have a working macOS VM and can debug this.

@ElijahPepe I tested the non capturing groups and they are not the perfect solution, when it comes to replacing in strings. In addition the replaced text has to be adjusted. Also a quick search showed that there are 28 lookbehind usages throughout the project. I'll get on it and keep you updated

**Comment by ElijahPepe** [comment]
I'm not an expert at Regex so I'll leave you to it. Feel free to change my PR if you'd like.

**Comment by ArthurHeitmann** [comment]
Small update:
All the regex problems were fixed in 2d1a580c8016bcccb2944483692a1ed73f8cfcaf and 8664e5d4fd2cf58a871ca8b60e00d9f2d155fa4c.

Now that the page is at least loading there's a new problem: Safari only allows *Autonomous custom elements* and not *Customized built-in elements* ([explanation](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements)). Fortunately only 3 classes are affected (`Ph_DropDownEntry`, `Ph_VoteButton`, and `Ph_BetterButton`). All 3 extend the `HTMLButtonElement`. This should be fixable though.

**Comment by ArthurHeitmann** [comment]
Everything seems to be fine now. I couldn't test audio on videos because of my VM. If there are any issues let me know.
